internal/oci/notation/notation.go

```diff
@@ -0,0 +1,319 @@
+	}
+
+	verifyOptions := notation.VerifyOptions{
+		ArtifactReference:    repoUrl,
+		MaxSignatureAttempts: 3,
+	}
+
+	_, outcomes, err := notation.Verify(ctx, *v.verifier, repo, verifyOptions)
+	if err != nil {
+		return oci.VerificationResultFailed, err
+	}
+
+	if len(outcomes) == 0 {
+		return oci.VerificationResultFailed, fmt.Errorf("signature verification failed for all the signatures associated with %s", url)
+	}
+
+	outcome := outcomes[0]
+
+	if outcome.VerificationLevel == trustpolicy.LevelSkip {
+		return oci.VerificationResultIgnored, nil
+	}
+
+	for _, i := range outcome.VerificationResults {
+		if i.Error != nil {
+			if i.Type == trustpolicy.TypeAuthenticity {
+				return oci.VerificationResultIgnored, i.Error
+			}
+
+			if i.Action == trustpolicy.ActionLog {
+				v.logger.Info(fmt.Sprintf("verification check for type %s failed for %s with message %s", i.Type, url, i.Error.Error()))
+			}
+		}
+	}
+
+	return oci.VerificationResultSuccess, nil
+}
+
+// remoteRepo is a function that creates a remote repository object for the given repository URL.
+// It initializes the repository with the provided URL and sets the PlainHTTP flag based on the value of the 'insecure' field in the Verifier struct.
+// It also sets up the credential provider based on the authentication configuration provided in the Verifier struct.
+// If authentication is required, it retrieves the authentication credentials and sets up the repository client with the appropriate headers and credentials.
+// Finally, it returns the remote repository object and any error encountered during the process.
+func (v *NotationVerifier) remoteRepo(repoUrl string) (*oras.Repository, error) {
+	remoteRepo, err := oras.NewRepository(repoUrl)
+	if err != nil {
+		return &oras.Repository{}, err
+	}
+
+	remoteRepo.PlainHTTP = v.insecure
+
+	credentialProvider := func(ctx context.Context, registry string) (oauth.Credential, error) {
+		return oauth.EmptyCredential, nil
+	}
+
+	auth := authn.Anonymous
+
+	if v.auth != nil {
+		auth = v.auth
+	} else if v.keychain != nil {
+		source := common.StringResource{Registry: repoUrl}
+
+		auth, err = v.keychain.Resolve(source)
+		if err != nil {
+			return &oras.Repository{}, err
+		}
+	}
+
+	if auth != authn.Anonymous {
+		authConfig, err := auth.Authorization()
+		if err != nil {
+			return &oras.Repository{}, err
+		}
+
+		credentialProvider = func(ctx context.Context, registry string) (oauth.Credential, error) {
+			if authConfig.Username != "" || authConfig.Password != "" || authConfig.IdentityToken != "" || authConfig.RegistryToken != "" {
+				return oauth.Credential{
+					Username:     authConfig.Username,
+					Password:     authConfig.Password,
+					RefreshToken: authConfig.IdentityToken,
+					AccessToken:  authConfig.RegistryToken,
+				}, nil
+			}
+			return oauth.EmptyCredential, nil
+		}
+	}
+
+	repoClient := &oauth.Client{
+		Header: http.Header{
+			"User-Agent": {"flux"},
+		},
+		Cache:      oauth.NewCache(),
```

Comment:
Removed the cache for now in cdeeb62

As you mentioned this needs a thorough testing on the cached bearer tokens, in case of azure managed identity tokens they are only valid for 1 hour, hence we will keep this change for future.